### DISCUSSION TITLE
Configuring/using versioning for each specific linter

### DISCUSSION THREAD

**Discussion by bwenstar** [golangci/golangci-lint#2173] [Q&A] [answered]
Hello,

I did look through the [documentation](https://golangci-lint.run/) but could not find anything.

I understand that each specific version of golangci-lint has different linters added(or potentially removed) however it does not specify which particular version of the linter that is is using.  

This question asks a very similar use-case that I am after: https://github.com/golangci/golangci-lint/issues/1661

golangci-lint itself is version controlled however if we want to use an older/newer linter is it possible for use to specify a fixed version through a configuration file?  Or at minimum, are we able to dump the versions of all linters that are used when running?

Thanks

**Comment by SVilgelm** [comment, marked as answer]
In most cases the change log contains the versions of the linters: https://golangci-lint.run/product/roadmap/#changelog

And it is impossible to set linter's version in the configuration and I'm not sure this will ever be possible.

**Reply by bwenstar** [reply]
Thanks for getting back to me.  Is it possible to get an output of the linter's versions through the application?  something like ```golangci-lint linters --version``` or similar?  That way if supplied with a .cfg and the known linters that are used, it is easy to get all linters used and their corresponding versions.

**Reply by ldez** [reply]
Hello, currently it's not possible but I worked on something to be able to do that (display the version of linters).
Before creating a PR, I need to understand the need behind this request.

**Reply by ldez** [reply]
But it will never be possible to choose a version of a linter.
